core/primitives/src/views.rs

```diff
@@ -1244,28 +1245,53 @@ impl From<ExecutionMetadata> for ExecutionMetadataView {
+                                Cost::ExtCost { .. } => "WASM_HOST_COST",
+                                Cost::WasmInstruction => "WASM_HOST_COST",
                             }
-                            Cost::ExtCost { ext_cost_kind: ext_cost } => {
-                                format!("{:?}", ext_cost).to_ascii_uppercase()
-                            }
-                            Cost::WasmInstruction => "WASM_INSTRUCTION".to_string(),
-                        },
-                        gas_used: profile_data[cost],
-                    })
-                    .collect();
+                            .to_string(),
+                            cost: match cost {
+                                // preserve old behavior that conflated some action
+                                // costs for profile (duplicates are removed afterwards)
+                                Cost::ActionCost {
+                                    action_cost_kind:
+                                        ActionCosts::deploy_contract_base
+                                        | ActionCosts::deploy_contract_byte,
+                                } => "DEPLOY_CONTRACT".to_owned(),
+                                Cost::ActionCost {
+                                    action_cost_kind:
+                                        ActionCosts::function_call_base
+                                        | ActionCosts::function_call_byte,
+                                } => "FUNCTION_CALL".to_owned(),
+                                Cost::ActionCost {
+                                    action_cost_kind:
+                                        ActionCosts::add_full_access_key
+                                        | ActionCosts::add_function_call_key_base
+                                        | ActionCosts::add_function_call_key_byte,
+                                } => "ADD_KEY".to_owned(),
+                                Cost::ActionCost {
+                                    action_cost_kind:
+                                        ActionCosts::new_action_receipt
+                                        | ActionCosts::new_data_receipt_base
+                                        | ActionCosts::new_data_receipt_byte,
+                                } => "NEW_RECEIPT".to_owned(),
+                                // other costs have always been mapped one-to-one
+                                Cost::ActionCost { action_cost_kind: action_cost } => {
+                                    format!("{:?}", action_cost).to_ascii_uppercase()
+                                }
+                                Cost::ExtCost { ext_cost_kind: ext_cost } => {
+                                    format!("{:?}", ext_cost).to_ascii_uppercase()
+                                }
+                                Cost::WasmInstruction => "WASM_INSTRUCTION".to_string(),
+                            },
+                            gas_used: profile_data[cost],
```

Comment:
Here, we don't need to go out of our way to preserve behavior, it's ok to add and remove entries to this table.

Reply:
Then what would you suggest? For example, I replace `new_receipt` with `new_action_receipt`, `new_data_receipt_base`, and `new_data_receipt_byte`. The database only has one number. The only way I can see this number make sense it to keep using the old name.

I want to add new entries for the new profile version but that will only happen a few PRs later. And this code here will still remain to handle the old data in the DBs.

Reply:
I don't suggest changing anything here, that's just a general note that, when we add two new costs, we can just drop the old one, we don't need to care about stuffing some dummy value there or something

Reply:
Oh I see, yes that makes sense!